tests/tests_integration/test_api/test_simulators/test_runs.py

```diff
@@ -129,35 +99,61 @@ async def test_run_with_wait_and_retrieve(
         assert created_run.id == retrieved_run.id
 
         logs_res = retrieved_run.get_logs()
-        assert logs_res is not None
-
         logs_res2 = cognite_client.simulators.logs.retrieve(ids=created_run.log_id)
-        assert logs_res2 is not None
 
+        assert logs_res is not None
+        assert logs_res2 is not None
         assert logs_res.dump() == logs_res2.dump()
 
         data_res = retrieved_run.get_data()
         data_res2 = cognite_client.simulators.runs.list_run_data(run_id=created_run.id)[0]
         assert data_res is not None
         assert data_res.dump() == data_res2.dump()
 
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "routine_external_id, routine_revision_external_id, model_revision_external_id",
+        [
+            (
+                None,
+                f"{RESOURCES.simulator_routine_external_id}_v1",
+                RESOURCES.simulator_model_revision_external_id,
+            ),
+            (
+                RESOURCES.simulator_routine_external_id,
+                None,
+                None,
```

Comment:
instead of `None, None` it would be cleaner to accept SimulationRunWrite as a parameter